sympy/integrals/transforms.py

```diff
@@ -1005,14 +1007,45 @@ def repl(ex, *args):
     expr = repl(expr, Unequality, replue)
     return S(expr)
 
+def expand_dirac_delta(expr):
+    """
+    Expand an expression involving DiractDelta to get it as a linear
+    combination of DiracDelta functions.
+    """
+    return _lin_eq2dict(expr, expr.atoms(DiracDelta))
 
 @_noconds
 def _laplace_transform(f, t, s_, simplify=True):
     """ The backend function for Laplace transforms. """
     from sympy import (re, Max, exp, pi, Min, periodic_argument as arg_,
-                       arg, cos, Wild, symbols, polar_lift)
+                       arg, cos, Wild, symbols, polar_lift, Add)
     s = Dummy('s')
-    F = integrate(exp(-s*t) * f, (t, 0, oo))
+    a = Wild('a', exclude=[t])
+    deltazero = 0
+    try:
+        integratable, deltadict = expand_dirac_delta(f)
+    except PolyNonlinearError:
+        raise IntegralTransformError(
+        'Laplace', f, 'could not expand DiracDelta expressions')
+    for dd in deltadict.keys():
+        if len(dd.args)>1 and dd.args[1]>0:
+            raise IntegralTransformError(
+                'Laplace', f, 'derivatives of DiracDelta not implemented')
+        argzero = dd.args[0].subs(t,0).is_zero
+        if argzero is True:
+            p = dd.args[0].match(a*t)
```

Comment:
Since we only match `DiracDelta(a*t)` I don't think that it is necessary to check `argzero`. Instead what we need to know is that `a` is not zero.

Reply:
If this matched the whole expression `DiracDelta(a*t)` then there also wouldn't be any need to check `len(dd.args)` above.

Reply:
OK, thanks!